app/scripts/modules/core/src/presentation/hooks/useDataSource.hook.ts

```diff
@@ -1,9 +1,9 @@
 import { useObservableValue } from './useObservableValue.hook';
 import { useLatestCallback } from './useLatestCallback.hook';
 
-import { ApplicationDataSource } from '../../application';
+import { ApplicationDataSource, IFetchStatus } from '../../application';
 
-export interface IDataSourceResult<T> {
+export interface IDataSourceResult<T> extends IFetchStatus {
   data: T;
```

Comment:
if you feel like getting really fancy you could parameterize `IFetchStatus` with `T = any` so everybody (even outside hooks) gets this optional stronger type. I don't think there's a pressing / important reason to do it right now though so feel free to ignore.